### ISSUE TITLE
[TensorRT-LLM][ERROR] Assertion failed: Request with beam width 1 differs from the max beam width 3

### ISSUE THREAD

**Issue by HarewVlad** [NVIDIA/TensorRT-LLM#1282]
I want to perform token healing, by submiting request with beam_size=1 to get generation logits for first token to select desired healed token. But I can't do that, because when i configure Executor with max_beam_size=3, seems like I can only send requests with beam_size=3

Is there a way I can send requests with beam_size=1 when Executor is configured with max_beam_size=3?

**Comment by mickaelseznec** [comment]
Hi @HarewVlad, unfortunately, an Executor started with max_beam_size=3 only accept requests with beam_size=3 for the moment.

**Comment by HarewVlad** [comment]
Hi @mickaelseznec. Thank you for reply, i'll wait for new updates then, it is not blocker for me for now.

**Comment by wxsms** [comment]
Is there any plan on this? i.e. set a bigger max_beam_size but request with a smaller one.

**Comment by njaramish** [comment]
I would also be very interested in this feature.
